api/types.proto

```diff
@@ -457,6 +463,8 @@ enum TaskState {
 	COMPLETE = 576 [(gogoproto.enumvalue_customname)="TaskStateCompleted"]; // successful completion of task (not error code, just ran)
 	SHUTDOWN = 640 [(gogoproto.enumvalue_customname)="TaskStateShutdown"]; // orchestrator requested shutdown
 	FAILED = 704 [(gogoproto.enumvalue_customname)="TaskStateFailed"]; // task execution failed with error
+	// TaskStateRejected means a task never ran, for instance if something about
+	// the environment failed (e.g. a network failed to allocate).
```

Comment:
@cyli in this example, but "network failed to allocate" you mean on the agent right? Maybe we should word this differently to avoid any confusion with the allocator.